src/hooks/resources/application.ts

```diff
@@ -79,12 +79,43 @@ sbvrUtils.addPureHook('POST', 'resin', 'application', {
 			throw new Error('App name may only contain [a-zA-Z0-9_-].');
 		}
 
-		return Promise.all([
-			deviceTypes
-				.normalizeDeviceType(request.values.device_type)
-				.then(deviceType => (request.values.device_type = deviceType)),
-			checkDependentApplication(args),
-		])
+		return Promise.try(() => {
+			if (
+				request.values.device_type != null &&
+				request.values.is_for__device_type == null
+			) {
+				// translate device_type to is_for__device_type
+				return deviceTypes
+					.getDeviceTypeIdBySlug(api, request.values.device_type)
+					.then(dt => {
+						if (!dt) {
+							throw new Error('Invalid device type value');
+						}
+						request.values.is_for__device_type = dt.id;
+					});
+			}
+		})
+			.then(() => {
+				if (!request.values.is_for__device_type) {
+					throw new Error('No device type specified for application');
+				}
+
+				return api.get({
+					resource: 'device_type',
+					id: request.values.is_for__device_type,
+					options: {
+						$select: ['slug'],
+					},
+				});
+			})
+			.then((dt: { slug: string }) => {
+				if (!dt) {
+					throw new deviceTypes.UnknownDeviceTypeError(
```

Comment:
There's also `InvalidDeviceTypeError` which I think may be more suitable as an error type to throw here -- `UnknownDeviceTypeError` is used with a `slug` as the message somewhere below. There's also a couple of plain `Error`s thrown above -- can't they be more specific? Do you mind going through the errors thrown and sort of making sure they make sense in each case? Are the response codes suitable for each case?